bokeh/models/plots.py

```diff
@@ -363,6 +364,38 @@ def add_tile(self, tile_source: TileSource, **kwargs: Any) -> TileRenderer:
             TileRenderer : TileRenderer
 
         '''
+        if not isinstance(tile_source, TileSource):
+            
+            if isinstance(tile_source, xyzservices.TileProvider):
+                selected_provider = tile_source
+
+            # allow the same string input you can now pass to get_provider
+            elif isinstance(tile_source, str):
+
+                flat_providers = xyzservices.providers.flatten()
+                all_providers = {k.lower().replace(".", ""): v for k,v in flat_providers.items()}
+
+                selected_provider = tile_source.lower().replace("_", "")
+                if "retina" in selected_provider:
+                    selected_provider = selected_provider.replace("retina", "")
+                    retina = True
+                if selected_provider not in all_providers.keys():
+                    raise ValueError('Unknown tile provider %s' % tile_source)
```

Comment:
```suggestion
                    raise ValueError(f'Unknown tile provider {tile_source}')
```
prefer f-strings throughout